### PR TITLE
AWS Lambda plugin documentation

### LINKED ISSUE #8654: [DOC] Add documentation for Lambda Processor and Sink for Dataprepper
**What do you want to do?**
Add documentation for Lambda Processor and Sink for Dataprepper
 
- [ ] Request a change to existing documentation
- [X] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** 
https://github.com/opensearch-project/data-prepper/issues/4699

**Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.
 Dataprepper 2.11

### PR DESCRIPTION
**Pull request by srikanthjg**
### Description
Adding documentation for aws lambda processor and sink

### Issues Resolved
Closes #8654 

### Version
Dataprepper 2.11

### Checklist
- [ X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+layout: default
+title: aws_lambda
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:3]:
```suggestion
title: AWS Lambda integration for Data Prepper
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+parent: Processors
+grand_parent: Pipelines
+nav_order: 10
+---
+
+# AWS Lambda Processor and Sink for Data Prepper
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:9]:
```suggestion
# AWS Lambda integration for Data Prepper
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+layout: default
+title: AWS Lambda integration for Data Prepper
+parent: Processors
+grand_parent: Pipelines
+nav_order: 10
+---
+
+# AWS Lambda integration for Data Prepper
+
+This document provides the configuration details and usage instructions for integrating AWS Lambda with Data Prepper, both as a processor and as a sink.
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:11]:
```suggestion
The AWS Lambda integration allows developers to use serverless computing capabilities within their Data Prepper pipelines for flexible event processing and data routing.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+nav_order: 10
+---
+
+# AWS Lambda integration for Data Prepper
+
+The AWS Lambda integration allows developers to use serverless computing capabilities within their Data Prepper pipelines for flexible event processing and data routing.
+
+## AWS Lambda processor configuration
+
+The `aws_lambda processor` enables invocation of an AWS Lambda function within your Data Prepper pipeline to process events. It supports both synchronous or asynchronous invocations based on your use case.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:15]:
```suggestion
The `aws_lambda processor` enables invocation of an AWS Lambda function within your Data Prepper pipeline to process events. It supports both synchronous and asynchronous invocations based on your use case.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+## Configuration fields
+
+You can configure the processor using the following configuration options:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:19]:
```suggestion
You can configure the processor using the following configuration options.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+Field                | Type    | Required | Description                                                                 
+-------------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:23]:
```suggestion
`function_name`      | String  | Required | The name of the AWS Lambda function to invoke.                               
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+The `aws_lambda processor` enables invocation of an AWS Lambda function within your Data Prepper pipeline to process events. It supports both synchronous or asynchronous invocations based on your use case.
+
+## Configuration fields
+
+You can configure the processor using the following configuration options:
+
+Field                | Type    | Required | Description                                                                 
+-------------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type`    | String  | Required | Specifies the invocation type; either `request-response` or `event`. Default is `request-response`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:24]:
```suggestion
`invocation_type`    | String  | Required | Specifies the invocation type, either `request-response` or `event`. Default is `request-response`.           
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+## Configuration fields
+
+You can configure the processor using the following configuration options:
+
+Field                | Type    | Required | Description                                                                 
+-------------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type`    | String  | Required | Specifies the invocation type; either `request-response` or `event`. Default is `request-response`.           
+`aws.region`         | String  | Required | AWS Region in which the Lambda function is located.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:25]:
```suggestion
`aws.region`         | String  | Required | The AWS Region in which the Lambda function is located.                         
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+## Configuration fields
+
+You can configure the processor using the following configuration options:
+
+Field                | Type    | Required | Description                                                                 
+-------------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type`    | String  | Required | Specifies the invocation type; either `request-response` or `event`. Default is `request-response`.           
+`aws.region`         | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`   | String  | Optional | ARN of the role to assume before invoking the Lambda function.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:26]:
```suggestion
`aws.sts_role_arn`   | String  | Optional | The Amazon Resource Name (ARN) of the role to assume before invoking the Lambda function.               
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+You can configure the processor using the following configuration options:
+
+Field                | Type    | Required | Description                                                                 
+-------------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type`    | String  | Required | Specifies the invocation type; either `request-response` or `event`. Default is `request-response`.           
+`aws.region`         | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`   | String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`        | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:27]:
```suggestion
`max_retries`        | Integer | Optional | The maximum number of retries for failed invocations. Default is `3`.             
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+You can configure the processor using the following configuration options:
+
+Field                | Type    | Required | Description                                                                 
+-------------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type`    | String  | Required | Specifies the invocation type; either `request-response` or `event`. Default is `request-response`.           
+`aws.region`         | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`   | String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`        | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.             
+`batch`              | Object  | Optional | Batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:28]:
```suggestion
`batch`              | Object  | Optional | The batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.                            
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+-------------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type`    | String  | Required | Specifies the invocation type; either `request-response` or `event`. Default is `request-response`.           
+`aws.region`         | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`   | String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`        | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.             
+`batch`              | Object  | Optional | Batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.                            
+`lambda_when`        | String  | Optional | Conditional expression to determine when to invoke the Lambda processor.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:29]:
```suggestion
`lambda_when`        | String  | Optional | A conditional expression that determines when to invoke the Lambda processor.     
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+`function_name`      | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type`    | String  | Required | Specifies the invocation type; either `request-response` or `event`. Default is `request-response`.           
+`aws.region`         | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`   | String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`        | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.             
+`batch`              | Object  | Optional | Batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.                            
+`lambda_when`        | String  | Optional | Conditional expression to determine when to invoke the Lambda processor.     
+`response_codec`     | Object  | Optional | Codec configuration for parsing Lambda responses. Default is `json`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:30]:
```suggestion
`response_codec`     | Object  | Optional |  A codec configuration for parsing Lambda responses. Default is `json`.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+`tags_on_match_failure` | List | Optional | List of tags to add to events when Lambda matching fails or encounters an unexpected error.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:31]:
```suggestion
`tags_on_match_failure` | List | Optional |  A list of tags to add to events when Lambda matching fails or encounters an unexpected error.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+          maximum_size: "5mb"
+          event_collect_timeout: PT10S
+      lambda_when: "event['status'] == 'process'"
+
+```
+{% include copy-curl.html %}
+
+## Usage
+
+The processor supports the following innovation types:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:60]:
```suggestion
The processor supports the following invocation types:
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+
+- `request-response`: Processor waits for Lambda function completion before proceeding.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:62]:
```suggestion
- `request-response`: The processor waits for Lambda function completion before proceeding.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+- `event`: Function is triggered asynchronously without waiting for a response.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:63]:
```suggestion
- `event`: The function is triggered asynchronously without waiting for a response.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+- `Batching`: When enabled, events are aggregated and sent in bulk to optimize Lambda invocations. Batch thresholds control event count, size limit, and timeout.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:64]:
```suggestion
- `Batching`: When enabled, events are aggregated and sent in bulk to optimize Lambda invocations. Batch thresholds control the event count, size limit, and timeout.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+- `Codec`: JSON is is used for both request and response codecs. Lambda must return JSON array outputs.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:65]:
```suggestion
- `Codec`: JSON is used for both request and response codecs. Lambda must return JSON array outputs.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+- `tags_on_match_failure`: Custom tags can be applied to events when Lambda processing fails or encounters unexpected issues.
+
+## Behavior
+
+When configured for batching, the AWS Lambda processor groups multiple events into a single request. This grouping is governed by batch thresholds, which can be based on event count, size limit, or timeout. The processor then sends the entire batch to the Lambda function as a single payload.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:70]:
```suggestion
When configured for batching, the AWS Lambda processor groups multiple events into a single request. This grouping is governed by batch thresholds, which can be based on the event count, size limit, or timeout. The processor then sends the entire batch to the Lambda function as a single payload.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 21]
```diff
@@ -0,0 +1,160 @@
...
+
+## Lambda response handling
+
+The `response_events_match` setting defines how Data Prepper handles the relationship between batch events sent to Lambda and the response received:
+
+- `true`: Lambda returns a JSON array with results for each batched event. Data Prepper maps this array back to its corresponding original event. , ensuring that each event in the batch gets the corresponding part of the response from the array.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:76]:
```suggestion
- `true`: Lambda returns a JSON array with results for each batched event. Data Prepper maps this array back to its corresponding original event, ensuring that each event in the batch gets the corresponding part of the response from the array.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 22]
```diff
@@ -0,0 +1,160 @@
...
+- `false`: Lambda returns one or multiple events for the entire batch. Response events are not correlated with the original events. Original event metadata is not preserved in the response events. For example, when `response_events_match` is set to `true`, the Lambda function is expected to return an equal number of response events as the orignal requests, maintaining the original order.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:77]:
```suggestion
- `false`: Lambda returns one or more events for the entire batch. Response events are not correlated with the original events. Original event metadata is not preserved in the response events. For example, when `response_events_match` is set to `true`, the Lambda function is expected to return the same number of response events as the number of original requests, maintaining the original order.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 23]
```diff
@@ -0,0 +1,160 @@
...
+
+## Limitations
+
+Note the following limitations:
+
+- Payload limitation: 6MB payload limit
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:83]:
```suggestion
- Payload limitation: 6 MB payload limit
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 24]
```diff
@@ -0,0 +1,160 @@
...
+Integration tests for this plugin are executed separately from the main Data Prepper build process. Use the following Gradle command to run these tests:
+
+```
+./gradlew :data-prepper-plugins:aws-lambda:integrationTest -Dtests.processor.lambda.region="us-east-1" -Dtests.processor.lambda.functionName="lambda_test_function"  -Dtests.processor.lambda.sts_role_arn="arn:aws:iam::123456789012:role/dataprepper-role
+```
+{% include copy-curl.html %}
+
+## AWS Lambda sink
+
+You can configure the sink using the following configuration options:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:97]:
```suggestion
You can configure the sink using the following configuration options.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 25]
```diff
@@ -0,0 +1,160 @@
...
+
+Field             | Type    | Required | Description                                                                 
+----------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`   | String  | Required | Name of the AWS Lambda function to invoke.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:101]:
```suggestion
`function_name`   | String  | Required | The name of the AWS Lambda function to invoke.                               
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+
+## AWS Lambda sink
+
+You can configure the sink using the following configuration options:
+
+Field             | Type    | Required | Description                                                                 
+----------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`   | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type` | String  | Optional | Specifies the invocation type. Default is `event`.             
+`aws.region`      | String  | Required | AWS Region in which the Lambda function is located.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:103]:
```suggestion
`aws.region`      | String  | Required | The AWS Region in which the Lambda function is located.                         
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 27]
```diff
@@ -0,0 +1,160 @@
...
+## AWS Lambda sink
+
+You can configure the sink using the following configuration options:
+
+Field             | Type    | Required | Description                                                                 
+----------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`   | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type` | String  | Optional | Specifies the invocation type. Default is `event`.             
+`aws.region`      | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`| String  | Optional | ARN of the role to assume before invoking the Lambda function.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:104]:
```suggestion
`aws.sts_role_arn`| String  | Optional | The ARN of the role to assume before invoking the Lambda function.               
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 28]
```diff
@@ -0,0 +1,160 @@
...
+
+You can configure the sink using the following configuration options:
+
+Field             | Type    | Required | Description                                                                 
+----------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`   | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type` | String  | Optional | Specifies the invocation type. Default is `event`.             
+`aws.region`      | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`| String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`     | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:105]:
```suggestion
`max_retries`     | Integer | Optional | The maximum number of retries for failed invocations. Default is `3`.             
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 29]
```diff
@@ -0,0 +1,160 @@
...
+You can configure the sink using the following configuration options:
+
+Field             | Type    | Required | Description                                                                 
+----------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`   | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type` | String  | Optional | Specifies the invocation type. Default is `event`.             
+`aws.region`      | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`| String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`     | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.             
+`batch`           | Object  | Optional | Batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:106]:
```suggestion
`batch`           | Object  | Optional | The batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.                              
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 30]
```diff
@@ -0,0 +1,160 @@
...
+
+Field             | Type    | Required | Description                                                                 
+----------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`   | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type` | String  | Optional | Specifies the invocation type. Default is `event`.             
+`aws.region`      | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`| String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`     | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.             
+`batch`           | Object  | Optional | Batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.                              
+`lambda_when`     | String  | Optional | Conditional expression to determine when to invoke the Lambda processor.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:107]:
```suggestion
`lambda_when`     | String  | Optional | A conditional expression that determines when to invoke the Lambda processor.          
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 31]
```diff
@@ -0,0 +1,160 @@
...
+----------------- | ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`   | String  | Required | Name of the AWS Lambda function to invoke.                               
+`invocation_type` | String  | Optional | Specifies the invocation type. Default is `event`.             
+`aws.region`      | String  | Required | AWS Region in which the Lambda function is located.                         
+`aws.sts_role_arn`| String  | Optional | ARN of the role to assume before invoking the Lambda function.               
+`max_retries`     | Integer | Optional | Maximum number of retries for failed invocations. Default is `3`.             
+`batch`           | Object  | Optional | Batch settings for the Lambda invocations. Default is `key_name = "events"`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.                              
+`lambda_when`     | String  | Optional | Conditional expression to determine when to invoke the Lambda processor.          
+`dlq`             | Object  | Optional | Dead-letter queue (DLQ) configuration for failed invocations.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:108]:
```suggestion
`dlq`             | Object  | Optional | A dead-letter queue (DLQ) configuration for failed invocations.                
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 32]
```diff
@@ -0,0 +1,160 @@
...
+      dlq:
+        region: "us-east-1"
+        sts_role_arn: "arn:aws:iam::123456789012:role/my-sqs-role"
+        bucket: "<<your-dlq-bucket-name>>"
+```
+{% include copy-curl.html %}
+
+## Usage
+
+The sink supports the following innovation types:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:137]:
```suggestion
The sink supports the following invocation types:
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 33]
```diff
@@ -0,0 +1,160 @@
...
+
+- `event`: Function is triggered asynchronously without waiting for a response.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:139]:
```suggestion
- `event`: The function is triggered asynchronously without waiting for a response.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 34]
```diff
@@ -0,0 +1,160 @@
...
+- `request-response`: Not supported for sink operations.
+- `Batching`: When enabled, events are aggregated and sent in bulk to optimize Lambda invocations. Default is `enabled`.
+- `DLQ`: Setup available for routing and precessing events that persistently fail Lamba invocations after multiple retry attempts.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:142]:
```suggestion
- `DLQ`:  A setup available for routing and processing events that persistently fail Lambda invocations after multiple retry attempts.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 35]
```diff
@@ -0,0 +1,160 @@
...
+
+## Advanced configurations
+
+The AWS Lambda processor and sink provide the following advanced options for secure function and performance optimization:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:146]: "secure function" => "security"?

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 36]
```diff
@@ -0,0 +1,160 @@
...
+- `event`: Function is triggered asynchronously without waiting for a response.
+- `request-response`: Not supported for sink operations.
+- `Batching`: When enabled, events are aggregated and sent in bulk to optimize Lambda invocations. Default is `enabled`.
+- `DLQ`: Setup available for routing and precessing events that persistently fail Lamba invocations after multiple retry attempts.
+
+## Advanced configurations
+
+The AWS Lambda processor and sink provide the following advanced options for secure function and performance optimization: 
+
+- IAM role assumption: Processor and sink support assuming the specified IAM role `aws.sts_role_arn` before Lambda invocation. This enhances secure handling by providing access control to AWS resources.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:148]:
```suggestion
- AWS Identity and Access Management (IAM) role assumption: The processor and sink support assuming the specified IAM role `aws.sts_role_arn` before Lambda invocation. This enhances secure handling by providing access control to AWS resources.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 37]
```diff
@@ -0,0 +1,160 @@
...
+- Concurrency management: When using the `event` invocation type, consider Lambda concurrency limits to avoid throttling.
+
+For further details on AWS Lambda integration with Data Prepper, see [AWS Lambda Documentation](https://docs.aws.amazon.com/lambda).
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:151]:
```suggestion
For more information about AWS Lambda integration with Data Prepper, see the [AWS Lambda documentation](https://docs.aws.amazon.com/lambda).
```

**Review by natebower — COMMENTED** [review]
@vagimeli @srikanthjg Please see my comments and changes and let me know if you have any questions. Thanks!

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 38]
```diff
@@ -0,0 +1,160 @@
...
+The sink supports the following invocation types:
+
+- `event`: The function is triggered asynchronously without waiting for a response.
+- `request-response`: Not supported for sink operations.
+- `Batching`: When enabled, events are aggregated and sent in bulk to optimize Lambda invocations. Default is `enabled`.
+- `DLQ`:  A setup available for routing and processing events that persistently fail Lambda invocations after multiple retry attempts.
+
+## Advanced configurations
+
+The AWS Lambda processor and sink provide the following advanced options for secure function and performance optimization:
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:146]:
```suggestion
The AWS Lambda processor and sink provide the following advanced options for security and performance optimization: 
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 39]
```diff
@@ -0,0 +1,160 @@
...
+- `DLQ`:  A setup available for routing and processing events that persistently fail Lambda invocations after multiple retry attempts.
+
+## Advanced configurations
+
+The AWS Lambda processor and sink provide the following advanced options for security and performance optimization: 
+
+- AWS Identity and Access Management (IAM) role assumption: The processor and sink support assuming the specified IAM role `aws.sts_role_arn` before Lambda invocation. This enhances secure handling by providing access control to AWS resources.
+- Concurrency management: When using the `event` invocation type, consider Lambda concurrency limits to avoid throttling.
+
+For more information about AWS Lambda integration with Data Prepper, see the [AWS Lambda documentation](https://docs.aws.amazon.com/lambda).
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:151]:
```suggestion
For more information about AWS Lambda integration with Data Prepper, see the [AWS Lambda Documentation](https://docs.aws.amazon.com/lambda).
```

**Comment by vagimeli** [comment]
> @vagimeli @srikanthjg Please see my comments and changes and let me know if you have any questions. Thanks!

@natebower Thank you for reviewing. I accepted your changes :)

**Review by vagimeli — APPROVED** [review]
Doc review completed and editorial feedback addressed

**Pull request merged**